doc/users/release_notes.rst

```diff
@@ -6,184 +6,9 @@
 Release notes
 =============
 
-.. include from another document so that it's easy to exclude this for releases
-.. include:: release_notes_next.rst
-
-3.4
```

Comment:
What is the benefit of splitting this page? AFAICS there is no difference in the rendering other than we have additional but pretty useless toc pages per version (https://63048-1385122-gh.circle-artifacts.com/0/doc/build/html/users/relnotes/3.4.html).

From a maintainance POV I think having everything in one file is simpler because one can more easily see what is there.

Reply:
Before doing this re-org the list on the left has nice accordion things 
![image](https://user-images.githubusercontent.com/199813/130548401-26641d6f-e36e-4ef9-b9b2-2f945664e5a6.png)
 instead of being one long list on my local build.  I'm using the latest pydata-sphinx-theme which apparently renders differently....